wandb/__init__.py

```diff
@@ -880,7 +881,9 @@ def init(job_type=None, dir=None, config=None, project=None, entity=None, reinit
     if dir:
         os.environ[env.DIR] = dir
         util.mkdir_exists_ok(wandb_dir())
-    if anonymous:
+    if anonymous is not None:
+        if anonymous not in ["allow", "must", "never"]:
+            raise ValueError("anonymous must be set to 'allow', 'must', or 'never'")
```

Comment:
This should be LaunchError
Is there an issue with how non valid params are used in the downstream code, or is this just to give the user better feedback to bad usage.  asking because you still have the possibility of the environment being set to invalid stuff.

Reply:
I mainly wanted to make sure that when users are calling `wandb.init` with invalid values for `anonymous` that they get a clean error message instead of us crashing or doing something unexpected.

I think we should have validation code that check the environment variable as well and crashes if it is invalid.